easybuild/framework/easyconfig/parser.py

```diff
@@ -114,9 +114,11 @@ def check_values_types(self, cfg):
         """
         wrong_type_msgs = []
         for key in cfg:
-            type_ok, _ = check_type_of_param_value(key, cfg[key])
+            type_ok, newval = check_type_of_param_value(key, cfg[key], auto_convert=True)
             if not type_ok:
                 wrong_type_msgs.append("value for '%s' should be of type '%s'" % (key, TYPES[key].__name__))
+            else:
```

Comment:
@Caylo: maybe this would be better?

```python
elif newval != cfg[key]:
    self.log.warning("Value for '%s' easyconfig parameter was converted from %s (type: %s) to %s (type: %s)",
                     key, cfg[key], type(cfg[key], newval, type(newval))
    cfg[key] = newval
```